test/whl-scale-tests/common.ps1

```diff
@@ -0,0 +1,170 @@
+        throw "number of upper/lower/numeric/special char must be lower or equal to length"
+    }
+    $uCharSet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    $lCharSet = "abcdefghijklmnopqrstuvwxyz"
+    $nCharSet = "0123456789"
+    $sCharSet = "{-)}^%(_!#$"
+    $charSet = ""
+    if($upper -gt 0) { $charSet += $uCharSet }
+    if($lower -gt 0) { $charSet += $lCharSet }
+    if($numeric -gt 0) { $charSet += $nCharSet }
+    if($special -gt 0) { $charSet += $sCharSet }
+    
+    $charSet = $charSet.ToCharArray()
+    $rng = New-Object System.Security.Cryptography.RNGCryptoServiceProvider
+    $bytes = New-Object byte[]($length)
+    $rng.GetBytes($bytes)
+ 
+    $result = New-Object char[]($length)
+    for ($i = 0 ; $i -lt $length ; $i++) {
+        $result[$i] = $charSet[$bytes[$i] % $charSet.Length]
+    }
+    $password = (-join $result)
+    $valid = $true
+    if($upper   -gt ($password.ToCharArray() | Where-Object {$_ -cin $uCharSet.ToCharArray() }).Count) { $valid = $false }
+    if($lower   -gt ($password.ToCharArray() | Where-Object {$_ -cin $lCharSet.ToCharArray() }).Count) { $valid = $false }
+    if($numeric -gt ($password.ToCharArray() | Where-Object {$_ -cin $nCharSet.ToCharArray() }).Count) { $valid = $false }
+    if($special -gt ($password.ToCharArray() | Where-Object {$_ -cin $sCharSet.ToCharArray() }).Count) { $valid = $false }
+ 
+    if(!$valid) {
+         $password = Get-RandomPassword $length $upper $lower $numeric $special
+    }
+    return $password
+}
+
+function SubstituteNameValuePairs {
+    param(
+        [string][Parameter(Mandatory=$true)] $InputFilePath,
+        [string][Parameter(Mandatory=$true)] $OutputFilePath,
+        [hashtable][Parameter(Mandatory=$true)] $Substitutions
+    )
+
+    # Ensure the input file exists
+    if (-not (Test-Path -Path $InputFilePath)) {
+        Write-Host "  Input File: '$InputFilePath' does not exist" -ForegroundColor Red
+        exit
+    }
+
+    $content = Get-Content $InputFilePath;
+    foreach($subItem in $Substitutions.GetEnumerator())
+    {
+        $content = $content.Replace($subItem.Name, $subItem.Value); 
+    }
+    $content | Set-Content $OutputFilePath;
+}
+
+<#
+.SYNOPSIS
+Get a temp directory
+.DESCRIPTION
+Gets path to a shared temporary directory for storing temporary files
+#>
+function Get-TempDir{
+    return $TempDir
+}
+
+<#
+.SYNOPSIS
+Build docker image
+.DESCRIPTION
+Builds the docker image for a log generator
+.PARAMETER image
+Full tag to use for the image in the format <uri>/<tag>:<version>
+.PARAMETER windowsVersion
+windows server core image version to use
+.PARAMETER buildDirectory
+Root build directory. This should be where the dockerfile is located.
+.EXAMPLE
+Build-DockerImage exampleacr.azurecr.io/generatelogs:latest ltsc2022 . 
+#>
+function Build-DockerImage {
+    param(
+        [Parameter(Mandatory = $true)][string] $imageTag,
+        [Parameter(Mandatory = $true)][string] $windowsVersion,

```

Comment:
Current state of the infra: We only deploy Windows Server 2022 right now.

If we are going to give the users option here, then we should also provide this option in infra deployment. 

If we think scaling testing on both is going to be a big difference, then we should consider testing, but cost will double.

Reply:
This has been addressed